### PR TITLE
python3: Use stable ABI

### LINKED ISSUE #325: Possible to link against python3.dll not python3x.dll (the one with specific minor version)
Python 3.12 is planed to release on Oct. 2th.

Some people want to keep using 3.11 because not enough exciting new features in 3.12
Some people would like to catch up with 3.12 because it's the latest.
Others would like to stick to 3.9 because many miniconda packages are for 3.9 only.

If this build can only depend on the general `Python3.dll`, people can choose what every they want.

And no need to update vim for new future python releases every year:

图片

### PR DESCRIPTION
**Pull request by k-takata**
* Compile against Python 3.12.0.
* Use Python stable ABI so that Vim can work with Python 3.8 or later.
* (Fix outdated comment about libsodium.)

Close #325

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ You can find those interperters here:
 * <del>[ActiveTcl](http://www.activestate.com/activetcl/downloads) 8.6.6</del> (currently disabled)
 * [LuaBinaries](http://luabinaries.sourceforge.net/download.html) 5.4
 * [Python](https://www.python.org/downloads/) 2.7
-* [Python 3](https://www.python.org/downloads/) 3.11
+* [Python 3](https://www.python.org/downloads/) 3.8 or later
```
Comment (k-takata) [inline README.md:34]: Do we need to explain Python stable ABI in README.md?

Comment (chrisbra) [reply README.md:34]: yes, at least mention that we are using the stable ABI now.

Comment (k-takata) [reply README.md:34]: Updated. Please let me know if you have better explanation.

**Comment by chrisbra** [comment]
thanks looks good now.

**Pull request merged**